main.nf

```diff
@@ -893,10 +893,13 @@ workflow.onComplete {
 
             focal_species_line = file("${configfile}").readLines().find{ it =~ /focal_species/ }
             if ( focal_species_line.strip().split("=").size() == 2 ) {
-                species_name = focal_species_line.split()[1].strip()
+                focal_species = focal_species_line.split()[1].strip()
+            }
+            else {
+                log.info "Focal species' name is not defined in the configuration file: possible leftovers in paralogs_distributions won't be deleted"
```

Comment:
Wouldn't this also be solved with a simple `focal_species = ""` instead?

Reply:
If we set it to empty string, the following line will match all folders within paralog_distributions, which in theory should be left untouched because they are part of other analyses. If we don't set it to empty string, it is null and the code will match no folder (which is what I wrote in the logging message above).

Reply:
Ah, you mean other wgd_species folders?! I didn't notice the '*' and didn't consider that there would be others (for the current ksrates version), but I guess yes, people may combine other analyses in the same folder which we probably shouldn't touch, I agree.

But what about removing the '*', then we should only deal with a folder `wgd_`, which should be the one we are aiming for? (Although I don't think the pipeline would (should) get that far anyways, unless we are dealing with (parts of) a previously existing analysis in which case the parlors folder would already have a species name...).

Reply:
If I remember well, the asterisk is not strictly needed to match the right paralog folder, but I added it so to have `paralog_dir_path` as a list, just like `ortholog_dir_path` is, and handle both of them the same way in the code below using `.each` (line 908), which wants an iterable object. It's more a trick.

Reply:
Alright, let's just go with the logging message you suggested, but maybe make it consistent with the others ones like it and use "Parameter focal_species in configuration file is empty: possible leftovers in paralogs_distributions won't be deleted". Also, should this be a log.error (or log.warn if exists) instead?

Reply:
Although we removed the other log line, I changed the log line as you suggested.
Here in the onComplete we are using only log.info (I'll check if we can even use log.error in this section).

Reply:
Using log.error in onComplete will give a white line! So I'll leave log.info.

Reply:
> Although we removed the other log line, I changed the log line as you suggested.

I meant consistent with Python messages from fc_configfile :)

Reply:
Uhm, right :)
I'll close the conversation!